numpy/core/fromnumeric.py

```diff
@@ -735,7 +735,7 @@ def partition(a, kth, axis=-1, kind='introselect', order=None):
     ================= ======= ============= ============ =======
        kind            speed   worst case    work space  stable
     ================= ======= ============= ============ =======
-    'introselect'        1        O(n)           0         no
+    'introselect'       O(n)     O(nlogn)        0         no
```

Comment:
Surprisingly, I think the worst case may actually be linear as well, because we're adopting an algorithm with a median of medians tree depth guard: https://en.wikipedia.org/wiki/Median_of_medians

`O(n)` worst case for the best selection implementations is also consistent with the Musser paper cited in the issue. At first, it might seem like the `1` speed notation is meant to represent `O(1)` constant time, but I suspect the NumPy devs didn't mess up that badly, and this is just for consistency with the sort ranking notations used for i.e., the sorting docs: https://numpy.org/doc/stable/reference/generated/numpy.sort.html (where `quicksort` has speed `1` to mean fastest on average I think, while other algos are numbered after that).

Maybe there is still an opportunity for clarification re: the "speed" column, but looking at the literature and what I see written in `numpy/core/src/npysort/selection.cpp`, the original linear worst case does seem correct.

Reply:
Makes sense.
We can close issue and this pr then.

Reply:
It would be very nice if we could then document this in the docstring, or at least point to the section of the code @tylerjereddy just mentioned, so that this doesn't raise questions again in the future. Would you be willing to do that @sapiosexual ? Thanks!

Reply:
I am not op of this issue so I am not sure I can contribute to the branch pr opened for, but sure.
If I can't I will just open another pr.

Let's revisit what we want to do:
1. Rename "speed" column to "speed ranking" or something, to make it more clear. Probably we should add column "average case" just to be sure no one confused it in future. Bonus track: add this column to the `numpy.sort`.
2. Clarify that it is not plain introselect but introselect with median of medians extension.

Am I correct?

Reply:
Ah sorry yes - perhaps @Kaushik-Iyer can get a chance first :)

Reply:
@sapiosexual Can you help me in doing the changes? As I'm a bit new to open source contributing, I would appreciate it